cmd/data-scanner.go

```diff
@@ -88,21 +86,15 @@ func initDataScanner(ctx context.Context, objAPI ObjectLayer) {
 	}()
 }
 
-type safeDuration struct {
-	sync.Mutex
-	t time.Duration
-}
+// safeDuration contains an atomic duration value.
+type safeDuration time.Duration
 
 func (s *safeDuration) Update(t time.Duration) {
-	s.Lock()
-	defer s.Unlock()
-	s.t = t
+	atomic.StoreInt64((*int64)(s), int64(t))
 }
 
 func (s *safeDuration) Get() time.Duration {
-	s.Lock()
-	defer s.Unlock()
-	return s.t
+	return time.Duration(atomic.LoadInt64((*int64)(s)))
 }
```

Comment:
Since we already use `go.uber.org/atomic`  we should just use `uatomic.NewDuration()` @anjalshireesh

Reply:
Done